### PR TITLE
ci: update release please to new GHA

### LINKED ISSUE #332: Expose SDK version as a module constant for use by providers/hooks
It would be useful to expose a module constant with the current SDK version so that providers can include it in User-Agent headers for example.

### PR DESCRIPTION
**Pull request by gruebel**
## This PR

- the `google-github-actions/release-please-action` GHA was deprecated and this the replacement 🙂 
- also added the `version.py` file, which hopefully will get updated correctly with the SDK version and can be used by 3rd parties

### Related Issues

Fixes #332

### CODE REVIEW

File: openfeature/version.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = "0.7.0"
```
Comment (beeme1mr) [inline openfeature/version.py:1]: What is this for?

**Review by beeme1mr — APPROVED** [review]

File: openfeature/version.py (hunk 1 above)
Comment (gruebel) [reply openfeature/version.py:1]: this gives the possibility to import the SDK version on runtime and add it to the user agent. nothing crucial for the release itself.

Comment (beeme1mr) [reply openfeature/version.py:1]: Should we have Release Please manage this?

Comment (gruebel) [reply openfeature/version.py:1]: Yeah, I think it should automatically do that, at least that's how I understood their documentation.

Comment (beeme1mr) [reply openfeature/version.py:1]: Okay, sounds good. Let's give it a shot.

**Pull request merged**